src/main/java/edu/indiana/dlib/amppd/service/impl/BundleServiceImpl.java

```diff
@@ -120,10 +158,78 @@ public Bundle deletePrimaryfiles(Long bundleId, Long[] primaryfileIds) {
 		Bundle bundle = bundleRepository.findById(bundleId).orElseThrow(() -> new StorageException("bundle <" + bundleId + "> does not exist!"));    		
 		if (primaryfileIds == null) {
 			log.warn("The given primaryfileIds is empty." );
+			return bundle;
 		}
 		for (Long primaryfileId : primaryfileIds) {
-			deletePrimaryfile(bundleId, primaryfileId);			
+			bundle = deletePrimaryfile(bundle, primaryfileId);			
+		}		
+		return bundle;
+	}
+
+	/**
+	 * @see edu.indiana.dlib.amppd.service.BundleService.updatePrimaryfiles(Long, String, Long[])
+	 */
+	@Override
+	public Bundle updateBundle(Long bundleId, String description, Long[] primaryfileIds) {
```

Comment:
Is there anything that prevents a duplicate bundle from being added?  I see when we search for bundles based on username/name, we only return one.  Should that instead return a list if multiple bundles with the same name can be created?

Reply:
At the moment the duplicate checking is done via front-end: when user tries to save, a request is sent to backend to check if a bundle with same name owned by him is already there, if so, he is asked to confirm to overwrite or cancel. We could enforce it at backend or even at DB level, and throw an exception if more than one is found, but that's kind a post-action. and is not supposed to be caused by user. After all, we still modify DB manually now and then, so there is no absolute garantee.